youtube_dl/extractor/iol.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+import re
+
+
+class IOLIE(InfoExtractor):
+    _VALID_URL = r'http://(tviplayer|(www\.tvi24))\.iol\.pt/.*/(?P<id>[0-9a-f]{24})($|\/)'
```

Comment:
There should be a lot of sites like that and not easy to enumerate them all.

Maybe just accept every *.iol.pt site?

Worst case scenario there's no video to extract...